### PR TITLE
add mms-infrastructure doc: import document from canonical

### PR DESCRIPTION
**Pull request by lduboeuf**
import document from https://wiki.ubuntu.com/Touch/Specs/MMSInfrastructure
modify mmsd to nuntium

### CODE REVIEW

File: systemdev/mms-infrastructure.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+MMS infrastructure components:
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:1]: without colon : please to stay consistent with other headlines

File: systemdev/mms-infrastructure.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+==============================
+
+
+**oFono [1]** - responsible for providing the data context used to transfer mms data (image/music). It also propagates wap push notifications to upper layers.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:5]: the footnotes look a bit out of place. can we make these into regular links?

Comment (doniks) [inline systemdev/mms-infrastructure.rst:5]: there's a mix of mms and MMS. could you make it MMS everywhere?

File: systemdev/mms-infrastructure.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+**nuntium [2]** - daemon that listens to wap push notifications and activates the mms data context on ofono on demand to send/receive mms's. It provides a local store, which will probably not be used by ubuntu-touch. The history service component will be used instead.
+
+**telepathy-ofono [3]**  - talks to nuntium through dbus and will be used to both relay messages from phone-app to nuntium and inject into the telepathy infrastructure mms's received by nuntium as multi-part messages [4]. This component also needs to mark messages as read and delete successfully received messages from nuntium.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:9]: "will be" again ... do you know whether it *is* used currently?

File: systemdev/mms-infrastructure.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+**history-service [5]**  - this component will watch the telepathy communication and store the messages received by telepathy-ofono or sent by phone-app.
+
+**telephony-service-approver [6]**  - this component is currently in charge of adding sms's to the messaging menu and to display incoming text notifications. It needs to be changed in order to add the received mms's to the messaging menu and display a notification that a new mms is available for download when auto-retrieval is disabled.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:13]: SMS

File: systemdev/mms-infrastructure.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+General Description
+-------------------
+
+
+Incoming MMS:
+^^^^^^^^^^^^^
+
+
+* An mms WAP push notification is received by ofono and propagated to nuntium
+* nuntium enables the mms context in ofono, retrieves the content from MMSC and propagates the new MMS to upper layers.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:42]: do you know what MMSC is?

File: systemdev/mms-infrastructure.rst (hunk 3 above)
Comment (lduboeuf) [reply systemdev/mms-infrastructure.rst]: i removed it, as it is used finally :)

File: systemdev/mms-infrastructure.rst (hunk 5 above)
Comment (lduboeuf) [reply systemdev/mms-infrastructure.rst]: i've added a link to the MMS infrastructure on Wikipedia where all is explained here

**Comment by lduboeuf** [comment]
should i update translation, or it will be done at merge time ?
Because pot seems out of sync now for other pages too

File: systemdev/mms-infrastructure.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
+MMS infrastructure components
+=============================
+
+`oFono <https://github.com/ubports/ofono>`_ - responsible for providing the data context used to transfer MMS data (image/music). It also propagates wap push notifications to upper layers.
+
+`nuntium <https://github.com/ubports/nuntium>`_ - daemon that listens to wap push notifications and activates the MMS data context on ofono on demand to send/receive MMS's. It provides a local store.
+
+`telepathy-ofono <https://github.com/ubports/telepathy-ofono>`_  - talks to nuntium through dbus and used to both relay messages from phone-app to nuntium and inject into the telepathy infrastructure MMS's received by nuntium as `multi-part messages <http://telepathy.freedesktop.org/spec/Channel_Interface_Messages.html#Mapping:Message_Part>`_. This component also mark messages as read and delete successfully received messages from nuntium.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:8]: and IS used 

markS

File: systemdev/mms-infrastructure.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+`history-service <https://github.com/ubports/history-service>`_ - this component watch the telepathy communication and store the messages received by telepathy-ofono or sent by messaging-app.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:10]: watchES

File: systemdev/mms-infrastructure.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+`telephony-service-approver <https://github.com/ubports/telephony-service>`_  - this component is in charge of adding SMS's to the messaging menu and to display incoming text notifications.
+
+`messaging-app <https://github.com/ubports/messaging-app>`_ - It render MMS's in the conversation view and also provide a way to attach media files to a message in order to send an MMS.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:14]: renderS

provideS

File: systemdev/mms-infrastructure.rst [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+Incoming MMS
+^^^^^^^^^^^^
+
+* An MMS WAP push notification is received by ofono and propagated to nuntium
+* nuntium enables the MMS context in ofono, retrieves the content from `MMSC <https://en.wikipedia.org/wiki/Multimedia_Messaging_Service>`_ and propagates the new MMS to upper layers.
+* A new MMS arrives at telepathy-ofono.
+* If automatic retrieval is not enabled or not available at the moment, a control message is sent to the app
+* messaging-app request the retrieval of the message to telepathy-ofono.
+* After the message is retrieved from MMSC, it is delivered to the apps via normal text channel in a multi-part message (one text/html part, one text/plain part and one part per attachment). If the control message was previously delivered, this real message have a flag telling it is a replacement for the previous message.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:27]: these real messages 


(or maybe: this real message ?? but I don't think so)

File: systemdev/mms-infrastructure.rst [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+* The new message history service save the attachment parts to disk, and replace the paths on text/html to point to the new attachment locations. It also save a flag and the content type.
```
Comment (doniks) [inline systemdev/mms-infrastructure.rst:28]: saveS

replaceS

saveS

File: systemdev/mms-infrastructure.rst (hunk 9 above)
Comment (lduboeuf) [reply systemdev/mms-infrastructure.rst:27]: not sure also, i would let `this`

**Pull request merged**